pkg/webui/console/containers/join-eui-prefixes-input/join-eui-prefixes-input.js

```diff
@@ -0,0 +1,161 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import classnames from 'classnames'
+import bind from 'autobind-decorator'
+import { defineMessages } from 'react-intl'
+
+import Select from '../../../components/select'
+import Input from '../../../components/input'
+
+import PropTypes from '../../../lib/prop-types'
+import computePrefix from './compute-prefix'
+
+import style from './join-eui-prefixes-input.styl'
+
+const m = defineMessages({
+  empty: 'No prefix',
+})
+
+const getOptions = prefixes => prefixes
+  .filter(prefix => Boolean(prefix.length))
+  .map(function ({ join_eui, length }) {
+    const computedPrefix = computePrefix(join_eui, length).toUpperCase()
+
+    return { label: computedPrefix, value: computedPrefix }
+  })
+
+const emptyOption = { label: m.empty, value: '' }
+
+@bind
+class JoinEUIPrefixesInput extends React.PureComponent {
+
+  constructor (props) {
+    super(props)
+
+    this.inputRef = React.createRef()
```

Comment:
Is the constructor necessary here?
```js
  inputRef = React.createRef()
  state = {
    …
  }
```
Saying this because you're doing it like this in `pkg/webui/components/input/byte.js` as well.